common/src/main/java/org/opensearch/ml/common/transport/memorycontainer/MLCreateMemoryContainerRequest.java

```
@@ -0,0 +1,108 @@
+        super(in);
+        this.mlCreateMemoryContainerInput = new MLCreateMemoryContainerInput(in);
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        super.writeTo(out);
+        this.mlCreateMemoryContainerInput.writeTo(out);
+    }
+
+    @Override
+    public ActionRequestValidationException validate() {
+        if (mlCreateMemoryContainerInput == null) {
+            return addValidationError("Memory container input can't be null", null);
+        }
+
+        ActionRequestValidationException exception = null;
+        SemanticStorageConfig semanticStorage = mlCreateMemoryContainerInput.getSemanticStorage();
+
+        if (semanticStorage != null && semanticStorage.isSemanticStorageEnabled()) {
+            if (semanticStorage.getModelType() == null) {
+                exception = addValidationError(SEMANTIC_STORAGE_MODEL_TYPE_REQUIRED_ERROR, exception);
+            } else {
+                FunctionName modelType = semanticStorage.getModelType();
+                if (modelType != FunctionName.TEXT_EMBEDDING && modelType != FunctionName.SPARSE_ENCODING) {
+                    exception = addValidationError(INVALID_MODEL_TYPE_ERROR, exception);
+                }
+
+                // Model ID is required when semantic storage is enabled
+                if (semanticStorage.getModelId() == null) {
+                    exception = addValidationError(SEMANTIC_STORAGE_MODEL_ID_REQUIRED_ERROR, exception);
+                }
+
+                // Validate dimension requirements
+                if (modelType == FunctionName.TEXT_EMBEDDING) {
+                    // Dimension is required for TEXT_EMBEDDING
+                    if (semanticStorage.getDimension() == null) {
+                        exception = addValidationError(TEXT_EMBEDDING_DIMENSION_REQUIRED_ERROR, exception);
+                    }
+                } else if (modelType == FunctionName.SPARSE_ENCODING) {
```

Do we need to throw this error? Can't we just ignore?